### PR TITLE
feat: Instantiate ReadBuffer as part of server creation

### PR DESCRIPTION
**Pull request by alamb**
Related to  #532 , builds on https://github.com/influxdata/influxdb_iox/pull/618

This PR, for the first time, instantiates a Read Buffer database instance in the server. 🎉 

At the moment, the `ReadBuffer` doesn't do anything other than look pretty (I haven't hooked it up to the write or query path yet) but that will be coming shortly. 

My basic plan is:
- [x] Implement SQL queries against `Databases` rather than as part of the Db instance itself (https://github.com/influxdata/influxdb_iox/pull/618)
- [x] Instantiate the read buffer (this PR)
- [ ] Add the basic way to load Chunk data from Arrow RecordBatches (@e-dard has a PR in this direction: https://github.com/influxdata/influxdb_iox/pull/616)
- [ ] Add basic support for load Chunk data from mutable store into the read buffer
- [ ] Hook up SQL queries against data in the read buffer and write tests against that (or maybe just the database??)

Administrivia

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: server/src/server.rs [hunk 1]
```diff
@@ -125,15 +126,17 @@ impl<M: ConnectionManager> Server<M> {
...
         let db_name = DatabaseName::new(db_name.into()).context(InvalidDatabaseName)?;
 
-        let buffer = if rules.store_locally {
+        let mutable_buffer = if rules.store_locally {
             Some(Arc::new(MutableBufferDb::new(db_name.to_string())))
         } else {
             None
         };
 
+        let read_buffer = Arc::new(ReadBufferDb::new());
```
Comment (alamb) [inline server/src/server.rs:135]: Here is the birthplace of the ReadBuffer in the server: 🎂

File: read_buffer/src/lib.rs [hunk 2]
```diff
@@ -15,199 +15,6 @@ use chunk::Chunk;
 use column::AggregateType;
 use row_group::{ColumnName, Predicate};
 
-/// The `Store` is responsible for providing an execution engine for reading
```
Comment (alamb) [inline read_buffer/src/lib.rs:18]: As I understand it the "store" notion is handled by the `DatabaseStore` trait and thus this (dead code) is not needed.

Comment (e-dard) [reply read_buffer/src/lib.rs:18]: This is going to conflict with my work in progress. Perhaps you could put this back and I could nuke it when I rebase my ongoing work?

**Review by e-dard — APPROVED** [review]
LGTM 👍

**Comment by alamb** [comment]
Build fails seems to be infra related: https://github.com/influxdata/influxdb_iox/pull/618/checks?check_run_id=1664010529

```
   Updating git repository `https://github.com/apache/arrow.git`
    Updating git submodule `https://github.com/apache/parquet-testing.git`
    Updating git submodule `https://github.com/apache/arrow-testing`
    Updating crates.io index
 Downloading crates ...
warning: spurious network error (2 tries remaining): failed to get 200 response from `https://crates.io/api/v1/crates/lazycell/1.3.0/download`, got 502
Warning: warning: spurious network error (1 tries remaining): failed to get 200 response from `https://crates.io/api/v1/crates/lazycell/1.3.0/download`, got 502
warning: spurious network error (2 tries remaining): failed to get 200 response from `https://crates.io/api/v1/crates/ring/0.16.19/download`, got 502
Warning: error: failed to download from `https://crates.io/api/v1/crates/lazycell/1.3.0/download`

Caused by:
  failed to get 200 response from `https://crates.io/api/v1/crates/lazycell/1.3.0/download`, got 502
Error: The process '/usr/local/cargo/bin/cargo' failed with exit code 101
```

**Pull request merged**